src/app/modules/group/components/associated-activity/associated-activity.component.ts

```diff
@@ -23,42 +26,51 @@ type ActivityId = string;
 })
 export class AssociatedActivityComponent implements OnDestroy, ControlValueAccessor {
 
-  rootActivity: null|{
-    id: ActivityId,
-    name: string|null,
-    path: string,
-  } = null;
+  rootActivity: NoActivity|NewActivity|ExistingActivity = { type: 'no-activity' };
+  rootActivityData: null|{name: string|null, path: string|null} = null;
 
   state: 'fetching'|'ready'|'error' = 'fetching';
 
-  private activityChanges = new Subject<{id: ActivityId|null, triggerChange: boolean}>();
+  allowedNewItemTypes: NewContentType<ItemType>[] = getAllowedNewItemTypes(false);
+
+  private activityChanges = new Subject<{activity: NoActivity|NewActivity|ExistingActivity, triggerChange: boolean}>();
+
+  private onChange: (value: NoActivity|NewActivity|ExistingActivity) => void = () => {};
 
-  private onChange: (value: ActivityId|null) => void = () => {};
+  searchFunction = (value: string): Observable<AddedContent<ItemType>[]> =>
+    this.searchItemService.search(value, [ 'Chapter', 'Course', 'Task' ]);
 
   constructor(
     private getItemByIdService: GetItemByIdService,
+    private searchItemService: SearchItemService,
   ) {
     this.activityChanges.pipe(
       switchMap(data => {
-        const id = data.id;
-        if (id === null) return of(readyState({ activity: null, triggerChange: data.triggerChange }));
+        if (!isExistingActivity(data.activity)) {
+          return of(readyState({
+            activity: data.activity,
+            activityData: isNewActivity(data.activity) ? { name: data.activity.name, path: null } : null,
+            triggerChange: data.triggerChange
+          }));
+        }
+
+        const id = data.activity.id;
+
         return merge(
           of(fetchingState()),
           this.getItemByIdService.get(id).pipe(map(item => readyState({
             triggerChange: data.triggerChange,
-            activity: {
-              id: id,
-              name: item.string.title,
-              path: incompleteItemStringUrl(id)
-            }
+            activity: { type: 'existing-activity', id: id } as ExistingActivity,
+            activityData: { name: item.string.title, path: incompleteItemStringUrl(id) },
```

Comment:
You have to handle errors!  For two reasons:
- you are working with a subject, so you are going to have it stuck if you have an error in a call
- there is no guarantee that the item is visible to the user... so you have to handle this case properly, for instance, it has to be an "Activity you do not have access to", so with no link but the right to remove it from this group.

Reply:
fixed, I added a lock icon next to the text to be more readable